src/analysis/convex/extreme.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import linear_algebra.affine_space.independent
+import algebra.big_operators.basic
+import analysis.convex.topology
+
+/-!
+# Extreme sets
+
+This file defines extreme sets and extreme points for set in a real normed space.
+
+## References
+
+See chapter 8 of [Convexity][simon2011]
+
+TODO:
+- add exposed sets to this file.
+- define convex independence and prove lemmas related to extreme points.
+-/
+
+open_locale classical affine
+open set
+
+variables {E : Type*} [add_comm_group E] [vector_space ℝ E] {x : E} {A B C : set E}
+
+/--
+A set B is extreme to a set A if B ⊆ A and all points of B only belong to (intrinsic) interiors of
+segments whose ends are in B.
+-/
+def is_extreme (A B : set E) :
+  Prop :=
+B ⊆ A ∧ ∀ x₁ x₂ ∈ A, ∀ x ∈ B, x ∈ segment x₁ x₂ → x₁ ≠ x → x₂ ≠ x → x₁ ∈ B ∧ x₂ ∈ B
```

Comment:
What are your thoughts on defining `segment.interior` so that this becomes
```suggestion
B ⊆ A ∧ ∀ x₁ x₂ ∈ A, ∀ x ∈ B, x ∈ (segment x₁ x₂).interior → x₁ ∈ B ∧ x₂ ∈ B
```

Reply:
I'm mixed feelings, because `(segment x₁ x₂).interior` is a particular instance of intrinsic interiors that I'm defining somewhere else, but at the same time `segment` is a useful enough instance of a convex hull on its own. I think I will do it.

Reply:
I think a good middle ground here is to define `open_segment`, which has the same meaning as intrinsic interior but is an algebraic notion rather than a topological one, and should be easy to work with. We can of course later show that the definitions are equivalent.

Reply:
Eheh, that's exactly what I was doing. I'm trying to get a decent API for it, hence the WIP label.